h2/src/main/org/h2/engine/Session.java

```diff
@@ -373,6 +373,8 @@ public void addLocalTempTable(Table table) {
      * @param table the table
      */
     public void removeLocalTempTable(Table table) {
+        // Exception thrown in org.h2.engine.Database.removeMeta if line below is missing with TestGeneralCommonTableQueries
+        database.lockMeta(this);
```

Comment:
I;m not seeing these exceptions when running that test - how do you reproduce this. Not keen to scatter more locking around unnecessarily

Reply:
You have to comment out the database.lockMeta(this); line shown and run the TestGeneralCommonTableQueries test:

Exception in thread "main" org.h2.jdbc.JdbcSQLException: General error: "java.lang.RuntimeException: Unexpected code path" [50000-196]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:295)
	at org.h2.message.DbException.toSQLException(DbException.java:268)
	at org.h2.message.TraceObject.logAndConvert(TraceObject.java:358)
	at org.h2.jdbc.JdbcPreparedStatement.execute(JdbcPreparedStatement.java:220)
	at org.h2.test.db.TestGeneralCommonTableQueries.testCreateTable(TestGeneralCommonTableQueries.java:378)
	at org.h2.test.db.TestGeneralCommonTableQueries.test(TestGeneralCommonTableQueries.java:42)
	at org.h2.test.db.TestGeneralCommonTableQueries.main(TestGeneralCommonTableQueries.java:26)
Caused by: java.lang.RuntimeException: Unexpected code path
	at org.h2.message.DbException.throwInternalError(DbException.java:242)
	at org.h2.message.DbException.throwInternalError(DbException.java:255)
	at org.h2.engine.Database.removeMeta(Database.java:945)
	at org.h2.table.Table.removeChildrenAndResources(Table.java:553)
	at org.h2.table.TableView.removeChildrenAndResources(TableView.java:422)
	at org.h2.engine.Session.removeLocalTempTable(Session.java:381)
	at org.h2.command.CommandContainer.stop(CommandContainer.java:130)
	at org.h2.command.Command.executeUpdate(Command.java:290)
	at org.h2.jdbc.JdbcPreparedStatement.execute(JdbcPreparedStatement.java:207)
	... 3 more